src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -144,7 +144,7 @@ bool ocl_engine::check_allocatable(const layout& layout, allocation_type type) c
     return true;
 }
 
-memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type type, bool reset) {
+memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type type, bool reset, size_t bytes_count) {
```

Comment:
Unused ?
``
memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type type, bool reset, size_t /*bytes_count*/) {
``

Reply:
No, I missed code to add bytes_count to `res = std::make_shared<ocl::gpu_usm>(this, layout, type, bytes_count);`.
I will update it. thanks.

Reply:
@ahnyoung-paul @yeonbok , do we really need this change in memory allocation API? With these changes it looks less intuitive and straightforward: usm_host allocation process behaves differently than other allocation types; and what if bytes_count and layout specify different amount of memory for some reason - we need to handle it as well.
As I can see, this new call is used only at constant op processing stage - could it be replaced with reinterpret_buffer call instead of API changes?

Reply:
I agree, thanks sergey.
@ahnyoung-paul  could you change like 
1. create constant data memory size with 1 for 0 size  layout
2. reinterprete that memory as 0

Reply:
Thank you for your comments

The reason of why I change bytes_count in memory is constant data with {0} layout . the actual bytes count is zero for {{0}, i32, bfyx} layout data but on the other hands {{}, i32, bfyx} has 32 bytes. so to allocate memory for {0} dimension layout, I changed bytes_count in memory.

I tried to change the code by @sshlyapn 's comments, but it is difficult to handle memory with {0} dimension. because it has zero bytes_count for the output of reinterpret_buffer from one dimension layout to zero dimension layout. it can't copy data and it will try zero size for host buffer in lock if bytes_count is zero. 

thus, I think it is hard to use reinterpret_buffer.

Reply:
mem.get_layout() is used in execution stage for dynamic loop / condition when loop/condition node set output memory buffer of inner body network into its output buffers.

Reply:
Discussed offline: 
The issue is caused by constant with {0} shape tensor and "0" size:
```
					<layer id="0" name="2627" type="Const" version="opset1">
						<data element_type="i64" shape="0" offset="54840154" size="0" />
						<output>
							<port id="0" precision="I64" names="2627">
								<dim>0</dim>
							</port>
						</output>
					</layer>
```
As I understand, in some cases such constant may be legal (possibly depends on context; need to clarify that), but in that case we should treat it as "empty constant tensor", so we are not allowed to read its value

Reply:
rollback previous commit about memory allocation function modification and modify code to create memory with empty constant {0} shape using reinterpret_buffer based on your suggestion. please review it again. thanks.